content/en/docs/concepts/services-networking/service.md

```diff
@@ -203,8 +203,6 @@ described in detail in [EndpointSlices](/docs/concepts/services-networking/endpo
 
 ### Application protocol
 
-{{< feature-state for_k8s_version="v1.19" state="beta" >}}
-
 The `AppProtocol` field provides a way to specify an application protocol for each Service port.
```

Comment:
As this is graduating to GA, can we mention the type of the values this field takes? (eg: an arbitrary string? one from a set of valid values?) Are there common values people might use?

Reply:
According to API spec,

```
This field follows standard Kubernetes label syntax. Un-prefixed names are reserved for IANA standard service names (as per RFC-6335 and http://www.iana.org/assignments/service-names). Non-standard protocols should use prefixed names such as 'mycompany.com/my-custom-protocol'. This is a beta field that is guarded by the `ServiceAppProtocol` feature gate and enabled by default.
```

According to the validation logic, this field must be a "qualified name".

Reply:
Ah, so we also need to move ServiceAppProtocol into the graduated section of https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

The paragraph from the API spec I would copy in verbatim (minus the beta details). That way, it can also get localized.

Reply:
Yes. The upstream PR is merged.

Reply:
@robscott would you be willing to expand the documentation? What's here now is rather light on detail.

Reply:
Isn't this field name `appProtocol` instead?

Reply:
@tengqm Depends on if you're using Go or yaml :). I'll update it to use `appProtocol` since yaml is likely more common.